tests/test_utils_url.py

```diff
@@ -123,14 +123,12 @@ def test_spaces(self):
         self.assertEqual(canonicalize_url("http://www.example.com/do?q=a%20space&a=1"),
                                           "http://www.example.com/do?a=1&q=a+space")
 
-    @unittest.skipUnless(six.PY2, "TODO")
     def test_normalize_percent_encoding_in_paths(self):
         self.assertEqual(canonicalize_url("http://www.example.com/a%a3do"),
-                                          "http://www.example.com/a%A3do"),
+                                          "http://www.example.com/a%A3do")
 
-    @unittest.skipUnless(six.PY2, "TODO")
     def test_normalize_percent_encoding_in_query_arguments(self):
-        self.assertEqual(canonicalize_url("http://www.example.com/do?k=b%a3"),
+        self.assertEqual(canonicalize_url("http://www.example.com/do?k=b%a3", encoding='latin1'),
```

Comment:
@kmike , latest commits allow not using `encoding` when normalizing, except when URL is passed as Unicode and has to be encoded (path and query).

I added a few tests on percent-encoded sequences that represent various trouble cases.

I added a test for raising an Exception when `encoding` is not compatible.
We could choose to normalize using UTF-8 in that case. What do you think?